app/src/main/java/com/gh4a/utils/StringUtils.java

```diff
@@ -139,4 +140,30 @@ public static SpannableStringBuilder applyBoldTags(Context context,
         }
         return ssb;
     }
+
+    @Nullable
+    public static int[] findMatchingLines(String input, String match) {
+        String[] lines = input.split("\n");
+        String[] matchLines = match.split("\n");
+
+        for (int i = 0; i < lines.length - matchLines.length; i++) {
+            boolean isMatch = true;
+            for (int j = 0; j < matchLines.length; j++) {
+                String cLine = lines[i + j];
+                String fLine = matchLines[j];
+                if ((j == 0 && !cLine.endsWith(fLine)) ||
+                        (j != 0 && j != matchLines.length - 1 && !cLine.equals(fLine)) ||
+                        (j == matchLines.length - 1 && !cLine.startsWith(fLine))) {
```

Comment:
Wouldn't this be easier to read:

```
boolean lineMatches =
        j == 0 ? cLine.endsWith(fLine) :
        j == matchLines.length - 1 ? cLine.startsWith(fLine) :
        cLine.equals(fLine);
if (!lineMatches) {
    isMatch = false;
    break;
}
```

Another naming suggestion: cLine -> inputLine, fLine -> matchLine